### ISSUE TITLE
C# and gdscript vector sign() methods handle 0s differently

### ISSUE THREAD

**Issue by d-eisenga** [godotengine/godot#48196]
**Godot version:**
3.3.stable.mono

**OS/device including version:**
Windows 10

**Issue description:**
According to [the documentation](https://docs.godotengine.org/en/3.3/classes/class_vector3.html#class-vector3-method-sign), `Vector2` and `Vector3`'s `sign()` method should return 1 for components with a value of 0.

This works as described in gdscript, but in c# the method returns zeros, not ones, for those vector components.

**Steps to reproduce:**
1. Attach the following gdscript to a scene node:
    ```gdscript
    extends Node
    func _ready():
    	print ("gdscript: %s" % Vector3.ZERO.sign());
    ```
2. Attach the following c# script to another scene node:
    ```c#
    using Godot;
    public class VectorTest : Node {
    	public override void _Ready () {
    		GD.Print($"C#: {Vector3.Zero.Sign()}");
    	}
    }
    ```
3. Run the scene, and see that the following is logged:
    ```
    C#: (0, 0, 0)
    gdscript: (1, 1, 1)
    ```

**Minimal reproduction project:**
[VectorTest.zip](https://github.com/godotengine/godot/files/6372564/VectorTest.zip)

**Comment by kleonc** [comment]
It's working like this since these methods were added in #31094.

C++ versions use this macro internally:
https://github.com/godotengine/godot/blob/dc98144b99f35cc6446e297e2c6082d8c441d540/core/typedefs.h#L121
C# versions call this method internally:
https://github.com/godotengine/godot/blob/dc98144b99f35cc6446e297e2c6082d8c441d540/modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs#L522-L531

---

The question is: which one is the expected behaviour?

**Comment by Chaosus** [comment]
cc @aaronfranke

**Comment by aaronfranke** [comment]
Duplicate of #44841, closing.

TL;DR: Already fixed in master, but this breaks compatibility so it can't be done in 3.x.

**Comment by kleonc** [comment]
@aaronfranke Just a note: some internal code could be using `SGN` macro with the assumption it's returning a non-zero value so the change made in #43903 could potentially introduce bugs... it might not be the case but I'd say all usages of that macro should have been verified prior to such change. Not sure if it has been done.
